### PR TITLE
feat(content): fruit products (jam, juice, cooked) will inherit name from component used

### PR DESCRIPTION
**Pull request by Lamandus**
## Purpose of change
Fruit jams, juices, pies, waffles, milkshakes, fruit slices, cookies, kompot, granola canned fruit and cooked fruit will now use the corresponding name of what it is made from.
If juice is made from apples it will be apple juice.
If jam is made from oranges, it will be orange jam.

Orange juice (oj) recipes and cranberry juice recipes got obsoleted due to being covered by the juice crafting.

the items themselves didn't get obsoleted. So dedicated cranberry or orange juice can still spawn.

(and egg names got corrected)

### CODE REVIEW

**Review by chaosvolt — DISMISSED** [review]
I had to check and confirm that yes, `sweet_fruit_fresh` and `sweet_fruit_irradiated` do indeed contain orange and cranberry, or else this PR would've needed that fixed. Looks good, just is a big-ass bunch of changes to sift through.

I wish component names could be an array though, so that you could condense the fresh and irradiated names into a single line, I assume that's not an option currently though.

**Comment by Lamandus** [comment]
> I had to check and confirm that yes, `sweet_fruit_fresh` and `sweet_fruit_irradiated` do indeed contain orange and cranberry, or else this PR would've needed that fixed. Looks good, just is a big-ass bunch of changes to sift through.
> 
> I wish component names could be an array though, so that you could condense the fresh and irradiated names into a single line, I assume that's not an option currently though.

I had the problem with peanut butter in my other PR, that lead to with butter, when using peanut butter. Maybe I can cut it down a bit.


And yes, orange and cranberry are in it.

**Comment by Lamandus** [comment]
yea, irritated can be cut out. I will correct it now, don't merge it yet

**Comment by chaosvolt** [comment]
Oh that's a fun one, you can mix the different types of juices and it only tracks one of the component names instead of just erasing it and calling it juice...
image

**Review by chaosvolt — APPROVED** [review]

**Comment by Lamandus** [comment]
Vat transformation does kill the fruit wine changes.
same with pulp, it is a byproduct, which doesn't inherit anything from the original.

**Comment by Lamandus** [comment]
should be good for now.

**Comment by Lamandus** [comment]
I am done, no more changes

File: data/json/items/comestibles/junkfood.json [hunk 1]
```diff
@@ -360,6 +360,34 @@
     "type": "COMESTIBLE",
     "id": "cookies",
     "name": "cookie",
+    "conditional_names": [
```
Comment (chaosvolt) [inline data/json/items/comestibles/junkfood.json:363]: Fruit cookies? o.O

**Review by chaosvolt — APPROVED** [review]

File: data/json/items/comestibles/junkfood.json (hunk 1 above)
Comment (Lamandus) [reply data/json/items/comestibles/junkfood.json:363]: Tasty

**Pull request merged**